File: CompilerRR/CompilerRR/src/ch/fhnw/cpib/compiler/classes/Parameter.java
package ch.fhnw.cpib.compiler.classes;

import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalFLOWMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IOptionalMECHMODE;
import ch.fhnw.cpib.compiler.cst.interfaces.IConcSyn.IStorageDeclaration;

public class Parameter implements IConcSyn.IParameter  {

	IConcSyn.IOptionalFLOWMODE flowMode;
	IConcSyn.IOptionalMECHMODE mechMode;
	IConcSyn.IStorageDeclaration stoDecl;
	public Parameter(IOptionalFLOWMODE flowMode, IOptionalMECHMODE mechMode, IStorageDeclaration stoDecl) {
		super();
		this.flowMode = flowMode;
		this.mechMode = mechMode;
		this.stoDecl = stoDecl;
	}
	
	
}

File: CompilerMR/IML_Compiler/IMLParser/DataTypes/Concrete/Parameter.cs
using IML_Compiler.IMLParser.DataTypes.Abstract;

namespace IML_Compiler.IMLParser.DataTypes.Concrete
{
    class Parameter : IParameter
    {
        public IOptionalFlowMode FlowMode { get; private set; }
        public IOptionalMechMode MechMode { get; private set; }
        public IStorageDeclaration StorageDeclaration { get; private set; }

        public Parameter (IOptionalFlowMode fMode, IOptionalMechMode mMode, IStorageDeclaration decl)
        {
            FlowMode = fMode;
            MechMode = mMode;
            StorageDeclaration = decl;
        }

        public AbsParameter ToAbySyn()
        {
            var abs = StorageDeclaration.ToAbySyn();
            return new AbsParameter()
            {
                StorageDeclaration = StorageDeclaration.ToAbySyn(),
                FlowMode = FlowMode.ToAbsSyn(),
                MechMode = MechMode.ToAbsSyn()
            };
        }
    }
}
